sled-agent/src/bootstrap/agent.rs

```diff
@@ -265,6 +294,28 @@ async fn sled_config_paths(storage: &StorageResources) -> Vec<Utf8PathBuf> {
         .collect()
 }
 
+async fn bootstore_fsm_state_paths(
+    storage: &StorageResources,
+) -> Vec<Utf8PathBuf> {
+    storage
+        .all_m2_mountpoints(sled_hardware::disk::CLUSTER_DATASET)
+        .await
+        .into_iter()
+        .map(|p| p.join(BOOTSTORE_FSM_STATE_FILE))
+        .collect()
+}
+
+async fn bootstore_network_config_paths(
+    storage: &StorageResources,
+) -> Vec<Utf8PathBuf> {
+    storage
+        .all_m2_mountpoints(sled_hardware::disk::CLUSTER_DATASET)
+        .await
+        .into_iter()
+        .map(|p| p.join(BOOTSTORE_NETWORK_CONFIG_FILE))
+        .collect()
+}
+
```

Comment:
I don't think it's currently an issue because of where you are *currently* calling these functions, but do we want these to return an error if the result is empty? Might be slightly more defensive.

(Admittedly, I did not check if the bootstore validated that the paths are non-empty too)

Reply:
The bootstore doesn't validate that they are non-empty. I wrote it like this copying the `sled_config_paths` above. I'm fine making all of these return an error and `expect` when used if you think it would make things safer.

Reply:
I think it would be worthwhile defensive coding, yeah. Would be nice to catch this early and explicitly.

Reply:
Fixed in 8f3d383dc